pkg/kubernetes/bootstrap.go

```diff
@@ -88,6 +108,34 @@ func BootstrapCluster(ctx *pulumi.Context) error {
 	return err
 }
 
+func deployPrometheusRemoteWriteBasicAuthSecret(ctx *pulumi.Context, cfg *config.Config, k8sConfig K8sPlatformConfigInput) (pulumi.Resource, error) {
+	if k8sConfig.ManagePrometheusRemoteWriteBasicAuthSecret {
+		username := ctx.Stack()
+		if k8sConfig.PrometheusRemoteWriteBasicAuthUsername != "" {
```

Comment:
Now that 1.18 is out, we could use https://github.com/samber/lo to do ternaries on these strings.

Reply:
hey, that's pretty cool. happy to see that pop not even a month after 1.18 was officially released with over 4k stars.

the if statement here is implemented in a bunch of places at the moment, so I think I'll keep with that for now